### PR TITLE
refactor icons.json

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- 🔥  `icons.json` name=>displayName, className=>iconName
  - `className` with `pt-icon-` prefix is a bunch of crufty data, esp given #2070 
- update `generate-icons-source`

#### Reviewers should focus on:

this PR should fix the broken Icons page in #2070 
new record for most files changed? `870`

### CODE REVIEW

File: packages/node-build-scripts/generate-icons-source [hunk 1]
```diff
@@ -79,9 +79,12 @@ async function writeLinesToFile(filename, ...lines) {
     fs.writeFileSync(outputPath, contents);
 }
 
-/** Removes `pt-icon-` prefix from icon className. */
-function toShortName(icon) {
-    return icon.className.replace("pt-icon-", "");
+/**
+ * Removes `pt-icon-` prefix from icon className.
```
Comment (adidahiya) [inline packages/node-build-scripts/generate-icons-source]: that's not what this does

File: packages/node-build-scripts/generate-icons-source [hunk 2]
```diff
@@ -111,13 +112,13 @@ function exportIconConsts(valueGetter) {
 async function buildPathsObject(objectName, size) {
     return Promise.all(
         ICONS_METADATA.map(async icon => {
-            const filepath = path.resolve(__dirname, `../../resources/icons/${size}px/${icon.className}.svg`);
+            const filepath = path.resolve(__dirname, `../../resources/icons/${size}px/${toLongName(icon)}.svg`);
```
Comment (adidahiya) [inline packages/node-build-scripts/generate-icons-source]: can we batch rename the SVG files too? no need for pt- in the filename

Comment (giladgray) [reply packages/node-build-scripts/generate-icons-source]: oof, sure

File: packages/node-build-scripts/generate-icons-source [hunk 3]
```diff
@@ -36,21 +36,21 @@ writeLinesToFile(
...
     ");",
 );
 
 // simple variable definitions
-writeLinesToFile("_icon-variables.scss", ...ICONS_METADATA.map(icon => `$${icon.className}: "${icon.content}";`));
+writeLinesToFile("_icon-variables.scss", ...ICONS_METADATA.map(icon => `$${toLongName(icon)}: "${icon.content}";`));
 
 // map ENUM_NAME to pt-icon-class-name
-writeLinesToFile("iconClasses.ts", ...exportIconConsts(icon => icon.className));
+writeLinesToFile("iconClasses.ts", ...exportIconConsts(toLongName));
```
Comment (giladgray) [inline packages/node-build-scripts/generate-icons-source]: @adidahiya thoughts on deleting this file as it's simply `Classes.iconClass(IconNames.AIRPLANE)`?

Comment (adidahiya) [reply packages/node-build-scripts/generate-icons-source]: hm yeah I guess that's fine... can you add it to the migration script?

**Comment by adidahiya** [comment]
are you going to delete `iconClasses.ts`?

**Comment by giladgray** [comment]
separate PR

**Pull request merged**